tests/components/azure_event_hub/test_hub.py

```diff
@@ -0,0 +1,228 @@
+"""The tests for the Azure Event Hub component."""
+from dataclasses import dataclass
+import logging
+from unittest.mock import patch
+
+from azure.eventhub.exceptions import EventHubError
+import pytest
+
+from homeassistant.components.azure_event_hub.const import (
+    CONF_MAX_DELAY,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DOMAIN,
+)
+from homeassistant.components.azure_event_hub.hub import AzureEventHub
+from homeassistant.const import STATE_ON
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+
+from .const import AZURE_EVENT_HUB_PATH, BASIC_OPTIONS, PRODUCER_PATH, SAS_CONFIG_FULL
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class FilterTest:
+    """Class for capturing a filter test."""
+
+    id: str
+    should_pass: bool
+
+
+async def _setup(hass, mock_call_later, filter_config):
+    """Shared set up for filtering tests."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    hass.data[DOMAIN] = {DATA_FILTER: FILTER_SCHEMA(filter_config)}
+    hub = AzureEventHub(hass, entry)
+    entry.add_to_hass(hass)
+    await hub.async_start()
+    await hass.async_block_till_done()
+    mock_call_later.assert_called_once()
+    return mock_call_later.call_args[0][2]
+
+
+async def _run_filter_tests(hass, tests, process_queue, mock_batch):
+    """Run a series of filter tests on azure event hub."""
+    for test in tests:
+        hass.states.async_set(test.id, STATE_ON)
+        await hass.async_block_till_done()
+        await process_queue(None)
+
+        if test.should_pass:
+            mock_batch.add.assert_called_once()
+            mock_batch.add.reset_mock()
+        else:
+            mock_batch.add.assert_not_called()
+
+
+@pytest.fixture(name="hub")
+async def setup_fixture(hass, mock_call_later):
+    """Create the setup."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    hass.data[DOMAIN] = {DATA_FILTER: FILTER_SCHEMA({})}
+    hub = AzureEventHub(hass, entry)
```

Comment:
Please don't interact with integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Set up the integration while patching the library appropriately.